source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -0,0 +1,296 @@
+            ::grpc::Status status = GetStub()->GetAttributeViReal64(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        std::string get_string_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViStringRequest request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViStringResponse response;
+            ::grpc::Status status = GetStub()->GetAttributeViString(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+    private:
+        std::shared_ptr<::grpc::Channel> channel_;
+        std::unique_ptr<::nidevice_grpc::Session> driver_session_;
+        std::unique_ptr<dmm::NiDMM::Stub> nidmm_stub_;
+        std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+        std::unique_ptr<dmm::NiDMMLibrary> nidmm_library_;
+        std::unique_ptr<dmm::NiDMMService> nidmm_service_;
+        std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiDMMDriverApiTest, NiDMMSelfTest_SendRequest_SelfTestCompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  dmm::SelfTestRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  dmm::SelfTestResponse response;
+
+  ::grpc::Status status = GetStub()->SelfTest(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+  EXPECT_EQ(0, response.self_test_result());
+  EXPECT_LT(0, strlen(response.self_test_message().c_str()));
+}
+
+TEST_F(NiDMMDriverApiTest, NiDMMReset_SendRequest_ResetCompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  dmm::ResetRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  dmm::ResetResponse response;
+
+  ::grpc::Status status = GetStub()->Reset(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiDMMDriverApiTest, NiDMMSetViReal64Attribute_SendRequest_GetViReal64AttributeMatches)
+{
+  const char* channel_name = "";
+  const dmm::NiDMMAttributes attribute_to_set = dmm::NiDMMAttributes::NIDMM_ATTRIBUTE_TRIGGER_DELAY;
+  const ViReal64 expected_value = 42.24;
+  ::grpc::ClientContext context;
+  dmm::SetAttributeViReal64Request request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_name);
+  request.set_attribute_id(attribute_to_set);
+  request.set_attribute_value(expected_value);
+  dmm::SetAttributeViReal64Response response;
+
+  ::grpc::Status status = GetStub()->SetAttributeViReal64(&context, request, &response);
+  
+  EXPECT_TRUE(status.ok());
```

Comment:
Line 215 and 216 should be part above block for consistency (and since they are associated with `SetAttributeViReal64` action you are doing in above block.

Reply:
made the change

Reply:
And then, there should be a blank line before `get_real64_attribute` to separate out "GetViReal64AttributeMatches" section of your test organization.

So essentially your test has 3 sections ->
1.  NiDMMSetViReal64Attribute
2. SendRequest
3. GetViReal64AttributeMatches

Blank lines in your code should separate these 3 sections for better readability.

Reply:
made that change